osdep/terminal-win.c

```diff
@@ -111,15 +111,35 @@ static bool is_native_out_vt(HANDLE hOut)
 void terminal_get_size(int *w, int *h)
 {
     CONSOLE_SCREEN_BUFFER_INFO cinfo;
-    HANDLE hOut = GetStdHandle(STD_OUTPUT_HANDLE);
+    HANDLE hOut = hSTDOUT;
     if (GetConsoleScreenBufferInfo(hOut, &cinfo)) {
         *w = cinfo.dwMaximumWindowSize.X - (is_native_out_vt(hOut) ? 0 : 1);
         *h = cinfo.dwMaximumWindowSize.Y;
     }
 }
 
+static bool get_font_size(int *w, int *h)
+{
+  CONSOLE_FONT_INFO finfo;
+  HANDLE hOut = hSTDOUT;
+  BOOL res = GetCurrentConsoleFont(hOut, FALSE, &finfo);
+  if (res) {
+      *w = finfo.dwFontSize.X;
+      *h = finfo.dwFontSize.Y;
+  }
+  return res;
+}
+
 void terminal_get_size2(int *rows, int *cols, int *px_width, int *px_height)
 {
+    int w = 0, h = 0, fw = 0, fh = 0;
+    terminal_get_size(&w, &h);
+    if (get_font_size(&fw, &fh)) {
+        *px_width = fw * w;
+        *px_height = fh * h;
+        *rows = w;
+        *cols = h;
+    }
```

Comment:
@na-na-hi 

This doesn't work well for recently added sixel support in conhost.

They use 10x20 sized cells (see https://github.com/microsoft/terminal/pull/17504).

Do you want to play around with it and improve it?

Reply:
> Not sure what's the best way to proceed from here.

Same. Would be nice to have it supported, but don't see easy path forward with this one.